src/weapon/w_013.c

```diff
@@ -5,7 +5,35 @@ INCLUDE_ASM("weapon/nonmatchings/w_013", func_5F000_8017A9CC);
 
 INCLUDE_ASM("weapon/nonmatchings/w_013", EntityWeaponAttack);
 
-INCLUDE_ASM("weapon/nonmatchings/w_013", func_ptr_80170004);
+void func_ptr_80170004(Entity* self) {
+    s16 temp_rotangle;
+    s16 temp_zpriority;
+
+    if (self->step == 0) {
+        if (self->ext.weapon.parent->entityId == 0) {
+            DestroyEntity(self);
+            return;
+        }
+        self->animCurFrame = self->ext.weapon.parent->animCurFrame;
+        self->animSet = self->ext.weapon.parent->animSet;
+        self->facingLeft = self->ext.weapon.parent->facingLeft;
+        self->unk5A = self->ext.weapon.parent->unk5A;
+        temp_zpriority = PLAYER.zPriority;
+        self->flags = 0x0C020000;
+        self->zPriority = temp_zpriority - 4;
+        self->palette = self->ext.weapon.parent->ext.generic.unk88.S16.unk0;
```

Comment:
Just to be clear, are you suggesting to create a new entry in `ext` to represent whatever is the parent of weapons?

Reply:
Do we know that the parent of a weapon is also a weapon? If so I'll make this change, but I'm just hesitant to use particular extensions when we don't know what a particular entity actually is.

Reply:
Understood. I decided for now to make unk88 be called ChildPalette, since it seems to be what the child (the `self` in this function) sets its palette based on.